src/test/java/io/neonbee/test/endpoint/odata/ODataBatchTest.java

```diff
@@ -0,0 +1,168 @@
+    void testBatchRequestSerialization(VertxTestContext testContext) {
+        testContext.verify(() -> {
+            String boundary = UUID.randomUUID().toString();
+            ODataRequest singleRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN);
+            ODataRequest batchRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN)
+                    .setBatch(boundary, singleRequest, singleRequest);
+
+            BatchParser batchParser = new BatchParser();
+            BatchOptions batchOptions = BatchOptions.with().isStrict(true).build();
+            ByteArrayInputStream inputStream = new ByteArrayInputStream(batchRequest.getBody().getBytes());
+            List<BatchRequestPart> partList =
+                    batchParser.parseBatchRequest(inputStream, boundary, batchOptions);
+            assertThat(partList).hasSize(2);
+
+            testContext.completeNow();
+        });
+    }
+
+    @ParameterizedTest(name = "{index}: with key {0}")
+    @MethodSource("withValidKeys")
+    @DisplayName("Read single existing entity via batch request")
+    void testReadExistingEntity(String id, VertxTestContext testContext) {
+        ODataRequest readRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN).setKey(id);
+        ODataRequest batchRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN)
+                .setBatch(readRequest);
+
+        assertODataBatch(requestOData(batchRequest), multipartResponse -> {
+            List<MultipartResponse.Part> parts = multipartResponse.getParts();
+            assertThat(parts).hasSize(1);
+
+            MultipartResponse.Part part = parts.get(0);
+            JsonObject acutal = part.getPayload().toJsonObject();
+            assertThat(acutal.getString("KeyPropertyString")).isEqualTo(id);
+
+            testContext.completeNow();
+        }, testContext);
+    }
+
+    @ParameterizedTest(name = "{index}: with key {0}")
+    @MethodSource("withInvalidKeys")
+    @DisplayName("Read single unkown entity via batch request")
+    void testReadUnknownEntity(String id, VertxTestContext testContext) {
+        ODataRequest readRequest = new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN).setKey(id);
+        ODataRequest batchRequest =
+                new ODataRequest(TestService1EntityVerticle.TEST_ENTITY_SET_FQN).setBatch(readRequest);
+
+        assertODataBatch(requestOData(batchRequest), multipartResponse -> {
+            // verify correct amount of response parts
+            List<MultipartResponse.Part> parts = multipartResponse.getParts();
+            assertThat(parts).hasSize(1);
+
+            // verify expected status code of response parts
+            MultipartResponse.Part part = parts.get(0);
+            assertThat(part.getStatusCode()).isEqualTo(404);
+
+            testContext.completeNow();
+        }, testContext);
+    }
```

Comment:
These two tests can be merged into one parameterized test, to reduce the duplicate code

Reply:
Good point. Will take care of that.